### PR TITLE
For the client binding propagate the full original completion [KVL-1112]

### PR DESCRIPTION
**Pull request by nicu-da**
Because more fields are added to the completion it gets more difficult to deconstruct it into our own models while keeping the same external API. Because of this we are propagating the full completion as well.

CHANGELOG_BEGIN

CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by fabiotudone-da — COMMENTED** [review]
Why is this needed at this stage? Doesn't this change the bindings API anyway?

**Comment by nicu-da** [comment]
> Why is this needed at this stage? Doesn't this change the bindings API anyway?

The only change to that is the removal of the extractors for the rejections.

I would say this is needed to propagate the completion to clients, as we enriched the completion with multiple fields, but those are not part of our internal models.

**Review by stefanobaghino-da — APPROVED** [review]

**Review by mziolekda — APPROVED** [review]

File: ledger/ledger-api-client/src/test/suite/scala/com/digitalasset/ledger/client/services/commands/CommandTrackerFlowTest.scala [hunk 1]
```diff
@@ -539,6 +549,8 @@ class CommandTrackerFlowTest
     }
 
   }
+  def failureCompletion(code: Code) =
```
Comment (miklos-da) [inline ledger/ledger-api-client/src/test/suite/scala/com/digitalasset/ledger/client/services/commands/CommandTrackerFlowTest.scala:552]: Nit:
```suggestion
  private def failureCompletion(code: Code): CompletionFailure =
```

File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala [hunk 2]
```diff
@@ -70,22 +70,18 @@ object CompletionResponse {
     response match {
```
Comment (miklos-da) [inline ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala:70]: Who depends on this method? Shouldn't we deprecate this?

File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala [hunk 3]
```diff
@@ -43,25 +47,21 @@ object CompletionResponse {
...
-      commandId: String,
-      transactionId: String,
-      originalStatus: StatusProto,
-  )
+      completion: Completion
+  ) {
 
+    lazy val commandId: String = completion.commandId
+    lazy val transactionId: String = completion.transactionId
+    lazy val originalStatus: StatusProto = completion.getStatus
```
[the remark this answers is not in the document]
Comment (S11001001) [reply ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala:55]:
```suggestion
    import completion.{commandId, transactionId, getStatus => originalStatus}
```

This should generally be done in all cases where you simply want to alias away stable property lookups.

File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala (hunk 2 above)
Comment (nicu-da) [reply ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala:68]: Good point. 
Canton is the dependent that actually needs this.

I will open a new PR once this is merged, which deprecates this usage.

File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicu-da) [reply ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala]: @S11001001 This is actually to keep the same external API so I don't think imports would work.
@stefanobaghino-da Done

File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala [hunk 4]
```diff
@@ -43,25 +47,21 @@ object CompletionResponse {
   private[daml] final case class QueueSubmitFailure(status: Status) extends TrackedCompletionFailure
 
   final case class CompletionSuccess(
-      commandId: String,
-      transactionId: String,
-      originalStatus: StatusProto,
-  )
+      completion: Completion
```
Comment (hubert-da) [inline ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala:50]: Nit: can probably be formatted to a one line

**Review by hubert-da — APPROVED** [review]
LGTM, just needs a conflict resolution

**Pull request merged**